### PR TITLE
PAINTROID-202 : Refactored dialogs (1)

### PR DESCRIPTION
**Pull request by FromG**
refactored the dialogs as described in the ticket.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by Electronix1337 — APPROVED** [review]
Lgtm. Everything works the same as on the develop branch.

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/AboutDialog.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.core.text.HtmlCompat
+import org.catrobat.paintroid.BuildConfig
+import org.catrobat.paintroid.R
+
+class AboutDialog : AppCompatDialogFragment() {
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return if (showsDialog) {
+            super.onCreateView(inflater, container, savedInstanceState)
+        } else inflater.inflate(R.layout.dialog_pocketpaint_about, container, false)
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/AboutDialog.kt:39]: please do note leave curly brackets

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/AboutDialog.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        aboutLicenseView.movementMethod = LinkMovementMethod.getInstance()
+        val catrobatUrl = getString(R.string.pocketpaint_about_url_catrobat,
+                getString(R.string.pocketpaint_about_url_catrobat_description))
+        aboutCatrobatView.text = HtmlCompat.fromHtml(catrobatUrl, HtmlCompat.FROM_HTML_MODE_LEGACY)
+        aboutCatrobatView.movementMethod = LinkMovementMethod.getInstance()
+    }
+
+    @SuppressLint("InflateParams")
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val inflater = activity!!.layoutInflater
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/AboutDialog.kt:65]: use requireActivity when ever possible

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/AboutDialog.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        val layout = inflater.inflate(R.layout.dialog_pocketpaint_about, null)
+        onViewCreated(layout, savedInstanceState)
+        return AlertDialog.Builder(context!!, R.style.PocketPaintAlertDialog)
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/AboutDialog.kt:68]: use requireContext when ever possible

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/AboutDialog.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                .setView(layout)
+                .setPositiveButton(R.string.done) { dialog, which -> dismiss() }
+                .create()
+    }
+
+    companion object {
+        @JvmStatic
+		fun newInstance(): AboutDialog {
+            return AboutDialog()
+        }
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/AboutDialog.kt:79]: change to fun newInstance(): AboutDialog = AboutDialog()

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/FeedbackDialog.kt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+                .setTitle(R.string.pocketpaint_rate_us_title)
+                .setPositiveButton(R.string.pocketpaint_ok)  { dialog, which -> dismiss() }
+                .create()
+    }
+
+    companion object {
+        @JvmStatic
+		fun newInstance(): FeedbackDialog {
+            return FeedbackDialog()
+        }
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/FeedbackDialog.kt:24]: same as aboutdialog

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/ImportImageDialog.kt [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                .setNegativeButton(R.string.pocketpaint_cancel) { dialog, which -> dismiss() }
+                .create()
+    }
+
+    companion object {
+        @JvmStatic
+		fun newInstance(): ImportImageDialog {
+            return ImportImageDialog()
+        }
+    }
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/ImportImageDialog.kt:43]: same as aboutdialog

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/IndeterminateProgressDialog.kt [hunk 7]
```diff
@@ -16,25 +16,20 @@
...
-		builder.setView(R.layout.pocketpaint_layout_indeterminate);
-		return builder.create();
-	}
-}
+    fun newInstance(context: Context?): AlertDialog {
+        val builder = AlertDialog.Builder(context!!, R.style.PocketPaintProgressDialog)
+        builder.setCancelable(false)
+        builder.setView(R.layout.pocketpaint_layout_indeterminate)
+        return builder.create()
+    }
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/IndeterminateProgressDialog.kt:34]:
```
return AlertDialog.Builder(context!!, R.style.PocketPaintProgressDialog)
        .setCancelable(false)
        .setView(R.layout.pocketpaint_layout_indeterminate)
        .create()
```

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatDialogFragment
+import org.catrobat.paintroid.R
+
+class InfoDialog : AppCompatDialogFragment(), DialogInterface.OnClickListener {
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val builder = AlertDialog.Builder(context!!, R.style.PocketPaintAlertDialog)
+        val arguments = arguments
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt:33]: not necassary to create a local variable

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        if (arguments != null) {
+            builder.setTitle(arguments.getInt(TITLE_RESOURCE_KEY))
+                    .setIcon(arguments.getInt(DRAWABLE_RESOURCE_KEY))
+                    .setMessage(arguments.getInt(MESSAGE_RESOURCE_KEY))
+        }
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt:38]: idiomatic kotlin would be

```
arguments?.let {
 ....
}
```

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        const val DRAWABLE_RESOURCE_KEY = "drawableResource"
+        const val MESSAGE_RESOURCE_KEY = "messageResource"
+        const val TITLE_RESOURCE_KEY = "titleResource"
+        @JvmStatic
+		fun newInstance(dialogType: DialogType, @StringRes messageResource: Int, @StringRes titleResource: Int): InfoDialog {
+            val infoDialog = InfoDialog()
+            val bundle = Bundle()
+            bundle.putInt(DRAWABLE_RESOURCE_KEY, dialogType.imageResource)
+            bundle.putInt(MESSAGE_RESOURCE_KEY, messageResource)
+            bundle.putInt(TITLE_RESOURCE_KEY, titleResource)
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt:63]: have a look at kotlin `apply()` function

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/LikeUsDialog.kt [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+                    dismiss()
+                }
+                .create()
+    }
+
+    companion object {
+        @JvmStatic
+		fun newInstance(): LikeUsDialog {
+            return LikeUsDialog()
+        }
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/LikeUsDialog.kt:30]: same as aboutdialog

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+        INFO(R.drawable.ic_pocketpaint_dialog_info), WARNING(R.drawable.ic_pocketpaint_dialog_warning);
+
+    }
+
+    companion object {
+        const val DRAWABLE_RESOURCE_KEY = "drawableResource"
+        const val MESSAGE_RESOURCE_KEY = "messageResource"
+        const val TITLE_RESOURCE_KEY = "titleResource"
+        @JvmStatic
+		fun newInstance(dialogType: DialogType, @StringRes messageResource: Int, @StringRes titleResource: Int): InfoDialog {
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt:58]: indentation does not look correct here

**Review by StanglTobias — CHANGES_REQUESTED** [review]
seems like you do use spaces for indentation. please change to tabs

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+	override fun onClick(dialog: DialogInterface, which: Int) {
+		dialog.cancel()
+	}
+
+	enum class DialogType(@get:DrawableRes
+						  @param:DrawableRes val imageResource: Int) {
+		INFO(R.drawable.ic_pocketpaint_dialog_info), WARNING(R.drawable.ic_pocketpaint_dialog_warning);
+
+	}
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt:51]: please reformat this

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+		fun newInstance(dialogType: DialogType, @StringRes messageResource: Int, @StringRes titleResource: Int): InfoDialog {
+			val infoDialog = InfoDialog()
+			val bundle = bundleOf(
+					DRAWABLE_RESOURCE_KEY to dialogType.imageResource,
+					MESSAGE_RESOURCE_KEY to messageResource,
+					TITLE_RESOURCE_KEY to titleResource
+			)
+			infoDialog.arguments = bundle
+			return infoDialog
+		}
```
Comment (StanglTobias) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/InfoDialog.kt:68]:
```
return InfoDialog().apply {
      arguments = bundleof( ... )
}
```

would be even nicer but you dont have to change it

**Review by StanglTobias — CHANGES_REQUESTED** [review]

**Review by StanglTobias — APPROVED** [review]

**Pull request merged**